### ISSUE TITLE
`@stylistic/selector-descendant-combinator-no-non-space` turns itself off for any selector carrying a comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#124]
Found by unskipping the first of the two skipped cases in `lib/rules/selector-descendant-combinator-no-non-space/index.test.js`, neither of which has an issue behind it.

### What minimal example or steps are needed to reproduce the bug?

```css
.foo /*comment*/  .bar {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/selector-descendant-combinator-no-non-space": true
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

The double space after the comment is reported and collapsed, the comment staying where the author put it:

```css
.foo /*comment*/ .bar {}
```

### What actually happened?

Nothing is reported. The rule bails out of any selector that has a comment anywhere in it:

```js
// Return early for selectors containing comments
// TODO: re-enable when parser and stylelint are compatible
if (selector.includes(`/*`)) return
```

One comment anywhere in a selector list turns the rule off for the whole rule node, so `.foo  .bar, .baz /*c*/ .qux {}` loses the warning on the first selector as well.

### Do you have a proposal to fix the bug?

Dropping the early return is not enough, because a comment inside a descendant combinator is not a node of its own: `postcss-selector-parser` folds the comment and the whitespace around it into the combinator's `value`. `.foo /*comment*/  /*comment*/ .bar` yields a single combinator whose `value` is the whole run, so the rule reports it once, with the run itself as the message argument and with the index of the run's start, and then declines to fix it, since the `/^\s+$/u` guard on the fix rejects anything but whitespace.

What the rule needs instead is to split the combinator's raw text into comments and whitespace runs, report each offending run at its own index, and rebuild the text with the comments kept and every run collapsed to a single space.

The skipped case can only be unskipped once that is written, and its expected warnings have to be rewritten with it — as they stand, they claim a single space at column 17 where the source has two, and a run at column 47 that falls inside a comment.

`@stylistic/selector-descendant-combinator-no-non-space` misses an illegal combinator for an unrelated reason, tracked separately.
